src/expression.c

```diff
@@ -118,7 +118,12 @@ Expression *getRightThis(Loc loc, Scope *sc, AggregateDeclaration *ad,
                     {
                         //printf("rewriting e1 to %s's this\n", f->toChars());
                         n++;
+#ifdef IN_GCC
+                        if (n > 1)
+                            e1 = new VarExp(loc, f->vthis);
+#else
                         e1 = new VarExp(loc, f->vthis);
+#endif
```

Comment:
> IIRC the 'this' lookup used to be entirely in the glue layer (and replicated in ctfe) and it was at least partially moved to where it should be, in semantic.

Most of invalid frame accesses are changed to front-end errors, but for other cases front-end would need more work.
I think the root of 14442 is that front-end does not have a way in AST to represent enclosing scope access via function frame pointer. It's also important to strictly handle closure creations (for @nogc) and optimize nested frame accesses (for scope and future enhanced escape analysis) in semantic analysis stage.